### PR TITLE
Complete OOP Exercise

### PR DESCRIPTION
**Pull request by deveLabR**
### OOP exercise

[Demo](https://develabr.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/deveLabR/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/develabr/oop/index.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  }
+  personInfoItems() {
+    return this.args.map((key) => this[key]).join('; ');
+  }
+}
+
+class Animal extends Creature {
+  constructor(species, name, gender, saying, friends) {
+    super(species, name, gender, saying, friends);
+    this.leg = 4;
```
Comment (OleksiyRudenko) [inline submissions/develabr/oop/index.js:18]: What makes `leg` so different from `name` or gender to handle `leg` property in a different way?

File: submissions/develabr/oop/index.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
+class Creature {
+  constructor(species, name, gender, saying, friends) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.saying = saying;
+    this.friends = friends;
+    this.args = ['species', 'name', 'gender', 'saying', 'friends'];
```
Comment (OleksiyRudenko) [inline submissions/develabr/oop/index.js:8]: Inhabitants are characterized rather by properties or features. No one says 'Name is my arg` or `Gender is my arg`

File: submissions/develabr/oop/index.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  constructor(species, name, gender, saying, friends) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.saying = saying;
+    this.friends = friends;
+    this.args = ['species', 'name', 'gender', 'saying', 'friends'];
+  }
+  personInfoItems() {
+    return this.args.map((key) => this[key]).join('; ');
```
Comment (OleksiyRudenko) [inline submissions/develabr/oop/index.js:11]: `key` is a generic, technical term.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@deveLabR pretty good job!
Let's make it more OOP compliant.
Also let's improve naming skill.

File: submissions/develabr/oop/index.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
+class Creature {
+  constructor(species, name, gender, leg, hands, saying, friends) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.leg = leg;
+    this.hands = hands;
```
Comment (OleksiyRudenko) [inline submissions/develabr/oop/index.js:7]: From the [requirements](https://kottans.org/documentation/docs/doc/code-review/#oop-exercise):
> Classes for species that do not have hands by natural design do not consequently have hands or any equivalent property

Subclasses inherit all properties from parent classes, which means they have it.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@deveLabR please revisit the requirements and education materials and make the code OOP-compliant.
Feel free asking questions here or in Students' chat.

**Comment by deveLabR** [comment]
Oops, I wanted it to be better, but it turned out as usual (( I'll fix it now

**Comment by OleksiyRudenko** [comment]
Use method overloading and `super` API. `super` can do more than just call parent's constructor.

**Comment by deveLabR** [comment]
I hope I understood correctly how to use overloading methods

**Review by OleksiyRudenko — APPROVED** [review]
@deveLabR great job!

**Pull request merged**